airbyte-integrations/connectors/source-hubspot/source_hubspot/components.py

```diff
@@ -137,3 +139,62 @@ def extract_records(self, response: requests.Response) -> Iterable[Mapping[str,
                             version["property"] = property_name
                             version[self.entity_primary_key] = primary_key
                             yield version | additional_keys
+
+
+class EngagementsHttpRequester(HttpRequester):
```

Comment:
Let's add a docstring here explaining why we need this custom retriever, why we have two endpoints, and a few warnings about its behavior.

The main warning that I noticed after reading the existing and new implementation is that when we switch over to the `/all` endpoint, we probably end up emitting about 10,000 duplicates. Our existing implementation might already do that so we don't necessarily need to fix this, but we should note it down here just in case

Reply:
added comment